src/script_pokemon_util_80A0058.c

```diff
@@ -0,0 +1,217 @@
+#include "global.h"
+#include "battle.h"
+#include "berry.h"
+#include "daycare.h"
+#include "event_data.h"
+#include "event_object_movement.h"
+#include "load_save.h"
+#include "malloc.h"
+#include "overworld.h"
+#include "party_menu.h"
+#include "pokedex.h"
+#include "script_pokemon_util_80A0058.h"
+#include "constants/items.h"
+#include "constants/species.h"
+
+static void CB2_ReturnFromChooseHalfParty(void);
+static void CB2_ReturnFromChooseBattleTowerParty(void);
+
+void HealPlayerParty(void)
+{
+    u8 i, j;
+    u8 ppBonuses;
+    u8 arg[4];
+
+    // restore HP.
+    for(i = 0; i < gPlayerPartyCount; i++)
+    {
+        u16 maxHP = GetMonData(&gPlayerParty[i], MON_DATA_MAX_HP);
+        arg[0] = maxHP;
+        arg[1] = maxHP >> 8;
+        SetMonData(&gPlayerParty[i], MON_DATA_HP, arg);
+        ppBonuses = GetMonData(&gPlayerParty[i], MON_DATA_PP_BONUSES);
+
+        // restore PP.
+        for(j = 0; j < MAX_MON_MOVES; j++)
+        {
+            arg[0] = CalculatePPWithBonus(GetMonData(&gPlayerParty[i], MON_DATA_MOVE1 + j), ppBonuses, j);
+            SetMonData(&gPlayerParty[i], MON_DATA_PP1 + j, arg);
+        }
+
+        // since status is u32, the four 0 assignments here are probably for safety to prevent undefined data from reaching SetMonData.
```

Comment:
What does the comment mean?

Reply:
speculation as to why `arg` is set this way